Adding static code analysis, adrs, documentation.

greggmarshall opened the pull request:
Adding the composer require steps for static code analysis, haven't added the configuration file(s) and set it up to scan the code on commits.

Also added the relevant Lullabot ADRs to docs/adrs.  We might want to change the deciders from the Lullabot employees listed to our team members.

jrosgiralt commented on docs/adrs/20210729-stage-file-proxy.md:
```diff
@@ -0,0 +1,55 @@
+  - Darren Petersen
+  - Marcos Cano
+  - Chris Albrecht
+
```
@greggmarshall as you mention in your notes, let's change the deciders according to our org. 

Can I suggest that you add the following deciders?

- Jordi Ros-Giralt
- Javier Prieto
- Gregg Marshall (if you think it's ok)

Thanks.

jrosgiralt reviewed (approved):
Code looks good and can be committed. Prior to committing, can you update the deciders list? See my note on the list of deciders to use. Thanks.

greggmarshall commented:
Updated ADRs with suggested deciders, also added Field Group module to support your config, finally updated core and modules to latest versions.  Generally I like to keep core and modules as up to date as possible during development to minimize later surprises.

jrosgiralt commented:
That's great, thanks for also adding field group module and updating core & modules.

jrosgiralt commented:
By the way, what composer commands did you use to update core and modules, these?

```
composer update drupal/core "drupal/core-*" --with-all-dependencies
composer update drupal/modulename --with-dependencies
```

greggmarshall commented:
Exactly.
